apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/Claimable.tsx

```diff
@@ -0,0 +1,646 @@
+                    </FormControl>
+                    <FormMessage />
+                  </FormItem>
+                )}
+              />
+            )}
+
+            <FormField
+              control={form.control}
+              name="primarySaleRecipient"
+              render={({ field }) => (
+                <FormItem className="flex-1">
+                  <FormLabel>Sale Recipient</FormLabel>
+                  <FormControl>
+                    <Input
+                      placeholder="0x..."
+                      {...field}
+                      disabled={!props.isOwnerAccount}
+                    />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+          </div>
+
+          <div className="grid grid-cols-1 gap-4 md:grid-cols-2">
+            <FormField
+              control={form.control}
+              name="pricePerToken"
+              render={({ field }) => (
+                <FormItem className="flex-1">
+                  <FormLabel>Price Per Token</FormLabel>
+                  <FormControl>
+                    <Input {...field} disabled={!props.isOwnerAccount} />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+            <FormField
+              control={form.control}
+              name="currencyAddress"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel>Currency</FormLabel>
+                  <CurrencySelector
+                    contractChainId={props.chainId}
+                    field={field}
+                  />
+                </FormItem>
+              )}
+            />
+          </div>
+
+          <div className="grid grid-cols-1 gap-4 md:grid-cols-2">
+            <FormField
+              control={form.control}
+              name="maxClaimableSupply"
+              render={({ field }) => (
+                <FormItem className="flex-1">
+                  <FormLabel>Max Available Supply</FormLabel>
+                  <FormControl>
+                    <Input {...field} disabled={!props.isOwnerAccount} />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            <FormField
+              control={form.control}
+              name="maxClaimablePerWallet"
+              render={({ field }) => (
+                <FormItem className="flex-1">
+                  <FormLabel>Maximum number of mints per wallet</FormLabel>
+                  <FormControl>
+                    <Input {...field} disabled={!props.isOwnerAccount} />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+          </div>
+
+          <FormField
+            control={form.control}
+            name="startTime"
```

Comment:
This is wrong - the form control here renders `DatePickerWithRange` which is controls both the `startTime` and `endTime` - use the `FormFieldSetup` component here instead